src/exchange_graph.cc

```diff
@@ -78,6 +78,7 @@ Arc::Arc(boost::shared_ptr<ExchangeNode> unode,
 Arc::Arc(const Arc& other)
     : unode_(other.unode()),
       vnode_(other.vnode()),
+      pref_(other.pref()),
```

Comment:
I'm not sure how this would be breaking anything. `pref_` is a double which is 0-valued by default. Copying should work in all cases.